packages/sitecore-jss/src/dictionary-service.ts

```diff
@@ -0,0 +1,85 @@
+import { AxiosResponse } from 'axios';
+import mcache from 'memory-cache';
+import { AxiosDataFetcher } from './data-fetcher';
+import { DictionaryPhrases, DictionaryServiceData } from './dataModels';
+import { HttpJsonFetcher } from './httpClientInterface';
+
+export type DictionaryServiceConfig = {
+  apiHost: string;
+  apiKey: string;
+  jssAppName: string;
+  /**
+   * Custom data fetcher
+   * @see HttpJsonFetcher<T>
+   */
+  dataFetcher?: HttpJsonFetcher<DictionaryServiceData>;
+  /**
+   * Enable/disable caching mechanism
+   * @default false
+   */
+  cacheEnabled?: boolean;
+  /**
+   * Cache timeout (ms)
+   * @default 10000
+   */
+  cacheTimeout?: number;
+};
+
+export class DictionaryService {
+  constructor(private dictionaryServiceConfig: DictionaryServiceConfig) {}
+
+  /**
+   * Fetch dictionary data
+   * @param {string} language
+   */
+  async fetchDictionaryData(language: string): Promise<DictionaryPhrases> {
+    const {
+      dataFetcher,
+      cacheTimeout = 10000,
+      cacheEnabled = false,
+    } = this.dictionaryServiceConfig;
+
+    const fetcher = dataFetcher || this.getDefaultFetcher();
+
+    const dictionaryServiceUrl = this.getUrl(language);
+
+    if (cacheEnabled) {
+      const cachedBody = mcache.get(dictionaryServiceUrl);
+
+      if (cachedBody) {
+        return cachedBody;
+      }
+    }
+
+    const response = await fetcher(dictionaryServiceUrl);
```

Comment:
I think we should expose and make use of the `fetchData` function that is part of `dataApi.ts`. Once the `fetchRouteData` and `fetchPlaceholderData` functions are fully removed, our layout-service code will need to be refactored to use this as well. This sets us up to consistently use the error checking and query parameter construction that it uses (to feed the dataFetcher / HttpJsonFetcher<T>).

Reply:
Good idea, I used `fetchData` in `dictionaryService`